charts/mlp/templates/_helpers.tpl

```diff
@@ -5,47 +5,42 @@
 Generated names
 */}}
 
-{{- define "mlp.resource-prefix-with-release-name" -}}
-    {{- $deployedChart := .Chart.Name -}}
-    {{- $chartVersion := .Chart.Version | replace "." "-" -}}
-    {{- $deployedReleaseName := .Release.Name -}}
-    {{ printf "%s-%s-%s"  $deployedChart $chartVersion $deployedReleaseName }}
-{{- end -}}
-
-{{- define "mlp.resource-prefix" -}}
-    {{- $deployedChart := .Chart.Name -}}
```

Comment:
can I suggest this: 

keep these resource prefix helper functions, and move the name-override if conditions from the mlp.name and mlp.fullname to these methods and use these resource prefix functions in all of the resources? 

That can make it look cleaner and self-descriptive instead of using mlp.name in creating resource names.

Reply:
Something like this? https://github.com/caraml-dev/helm-charts/pull/109/commits/46ce923f2dcfb9b19751e231fa100449bae88340

Reply:
This is ok too. Although I was suggesting resource-prefix be used in mlp.name, instead of the opposite.